### DISCUSSION TITLE
pvcell _VocSTC() and  Voc() methods

### DISCUSSION THREAD

**Discussion by lp74** [SunPower/PVMismatch#148] [Q&A]
Hi @mikofski,

this is a great library, thank you very much for your effort. 
I've got a question about the open circuit voltage as computed in pvcell.py:

```py
C = self.Aph * self.Isc + self.Isat1 + self.Isat2
delta = self.Isat2 ** 2. + 4. * self.Isat1 * C
return self.Vt * np.log(((-self.Isat2 + np.sqrt(delta)) / 2. / self.Isat1) ** 2.)
```

May you give some references (article) for this estimation? I am a little bit lost here.

**Comment by lp74** [comment]
I've understood.

It is very simple: you solve a second order equation in x = sqrt(Voc/Vt).

Thank you anyway.

**Reply by mikofski** [reply]
You beat me to it! Yes you just consider an ideal cell with `Rs=0` and `Rsh=Inf`, then solve. I was going to say it's similar to the 1- diode equivalent explained in [pvlib single diode section on Bishop's method](https://pvlib-python.readthedocs.io/en/latest/singlediode.html#bishop-s-algorithm), except for 2 diodes. Sorry I don't know a reference,  but it's a common solar trick
